src/contracts/collections/pool_keys.rs

```diff
@@ -38,7 +38,7 @@ impl PoolKeys {
         match self.get_index(pool_key) {
             Some(index) => {
                 self.pool_keys_by_index.remove(index);
-                self.pool_keys_length -= 1;
+                self.pool_keys_length = self.pool_keys_length.checked_sub(1).unwrap();
```

Comment:
The possible error of checked operation should be checked. In all of the places I would add additional errors
```
InvariantError::AddOverflow
InvariantError::SubUnderflow
InvariantError::MulOverflow
InvariantError::DivOverflow
```